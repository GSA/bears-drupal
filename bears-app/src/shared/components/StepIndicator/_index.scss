.step-back-link {
  padding-bottom: 30px;
}
