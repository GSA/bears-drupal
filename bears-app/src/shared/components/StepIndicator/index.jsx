import { StepBackLink } from '../index'
import PropTypes from 'prop-types'

/**
 * a functional component that renders a step indicator for our form sections
 * @component
 * @param {array} data - an array of sections
 * @param {boolean} noHeadings - determinate to render headings or not
 * @param {number} current - inherited state, indicates index value
 * @param {boolean} setCurrent - inherited function to mangae index value
 * @param {string} backLinkLabel - inherited value for back link value
 * @param {func} handleCheckRequiredFields - inherited handler
 * @return {html} returns markup for a usa step indicator
 */
const StepIndicator = ({
  data,
  noHeadings,
  current,
  setCurrent,
  backLinkLabel,
  handleCheckRequriedFields,
}) => {
  /**
   * a functional component that supports a11y for completed steps
   * @component
   * @param {boolean} completed - assigns if the step has been completed
   * @param {boolean} noHeadings - determinate to render headings or not
   * @return {html} returns markup for a usa step indicator
   */
  const CompletedSR = ({ completed }) => (
    <span className="usa-sr-only">{completed}</span>
  )

  /**
   * a functional component that creates our step in the list
   * @component
   * @param {string} heading - associated with the step link
   * @param {number} current - the current index from state
   * @param {function} setCurrent - updates the current index from onClick
   * @param {boolean} completed - state of completion
   * @param {number} index - the index of this item
   * @return {html} returns markup for a usa step indicator segment
   */
  const StepIndicatorSegment = ({
    heading,
    current,
    setCurrent,
    completed,
    index,
    handleCheckRequriedFields,
  }) => {
    const statusClass = current === index ? '--current' : ''
    return (
      <li
        className={`usa-step-indicator__segment usa-step-indicator__segment${statusClass}`}
        aria-current={completed}
        onClick={() =>
          completed === false &&
          handleCheckRequriedFields() === true &&
          setCurrent(index + 1)
        }
      >
        <span className="usa-step-indicator__segment-label">
          {!noHeadings && heading} <CompletedSR status={completed} />
        </span>
      </li>
    )
  }

  return (
    <div>
      <div className="usa-step-indicator" aria-label="progress">
        <ol className="usa-step-indicator__segments">
          {data &&
            data.map((step, i) => {
              const heading = step.section.heading
              return (
                <StepIndicatorSegment
                  heading={heading}
                  key={`${heading}-${i}`}
                  index={i}
                  current={current}
                  setCurrent={setCurrent}
                  completed={false}
                  handleCheckRequriedFields={handleCheckRequriedFields}
                />
              )
            })}
        </ol>
      </div>
      <StepBackLink currentIndex={current} setCurrent={setCurrent}>
        {backLinkLabel}
      </StepBackLink>
    </div>
  )
}

StepIndicator.propTypes = {
  data: PropTypes.array,
  noHeadings: PropTypes.bool,
  current: PropTypes.number,
  setCurrent: PropTypes.func,
  backLinkLabel: PropTypes.string,
  handleCheckRequiredFields: PropTypes.func,
}

export default StepIndicator
