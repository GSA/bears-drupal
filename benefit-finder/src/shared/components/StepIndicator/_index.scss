@use '../../styles/colors/_index.scss' as color;

#benefit-finder {
  .step-back-link {
    padding-bottom: 30px;
    color: color.$medium-grey;
  }
}
