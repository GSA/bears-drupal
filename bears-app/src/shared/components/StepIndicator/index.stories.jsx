import StepIndicator from './index.jsx'
import content from '../../api/mock-data/content-data.js'

const { lifeEventForm } = JSON.parse(content)
const data = lifeEventForm.sectionsEligibilityCriteria

export default {
  component: StepIndicator,
  tags: ['autodocs'],
  args: {
    data,
  },
}

export const Primary = {}

export const NoHeadings = {
  args: {
    ...Primary,
    noHeadings: true,
  },
}
