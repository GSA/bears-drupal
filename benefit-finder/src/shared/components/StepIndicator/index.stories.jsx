import StepIndicator from './index.jsx'
import content from '../../api/mock-data/content-data.js'

const { lifeEventForm } = JSON.parse(content)
const data = lifeEventForm.sectionsEligibilityCriteria

export default {
  component: StepIndicator,
  tags: ['autodocs'],
  args: {
    data,
    current: 0,
  },
}

export const Primary = {}

export const NoHeadings = {
  args: {
    ...Primary,
    noHeadings: true,
  },
}
