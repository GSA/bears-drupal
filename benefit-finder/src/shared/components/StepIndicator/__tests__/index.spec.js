import renderer from 'react-test-renderer'
import StepIndicator from '../index.jsx'

describe('StepIndicator', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(<StepIndicator />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
