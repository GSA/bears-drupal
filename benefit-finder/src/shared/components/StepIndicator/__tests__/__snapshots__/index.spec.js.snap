// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepIndicator renders a match to the previous snapshot 1`] = `
<div>
  <div
    aria-label="progress"
    className="usa-step-indicator"
    tabIndex={0}
  >
    <ol
      className="usa-step-indicator__segments"
    />
  </div>
  <button
    className=""
    onClick={[Function]}
    type="button"
  >
    Back
  </button>
</div>
`;
